src/sql/src/normalize.rs

```diff
@@ -349,6 +349,19 @@ pub fn create_statement(
                     }
                 }
             }
+            if let mz_sql_parser::ast::CreateSourceConnector::Kafka(KafkaSourceConnector {
+                connector: KafkaConnector::Reference { connector },
+                ..
+            }) = connector
+            {
+                *connector = unresolve(
```

Comment:
This series of calls seems off to me--doesn't seem like anything else in this module makes quite the same series of nested calls. Is `connector`'s type right, i.e. should we handle its parsing as some other type that requires fewer hoops?

Reply:
This is the exact same set of calls used to resolve the name in the "tx_metadata" with option logic, I stole the approach from there in fact since it is doing an identical thing just in a different place. This is the block I added to solve the problem that sources would create but then restarting mz would fail with catalogue corruption because we didn't fully qualify the name before persisting the source.

That said, I am very open to simplifying if there is a better way to allow for referencing a connector by name without requiring the name be fully qualified by the user.

Reply:
This does indeed look to work so I will change both call sites. Thanks!